update-testsuite.py

```diff
@@ -0,0 +1,203 @@
+        self.upstream = upstream
+
+    def __str__(self):
+        repo = f'WebAssembly/{self.repo}'
+        if self.branch != 'main':
+            repo += f'@{self.branch}'
+        return repo
+
+    # Fetch the latest revision from this repository and store the revision
+    # within `self` of what was found.
+    def update(self):
+        self.git('fetch', self.url(), self.branch)
+        self.rev = self.git('rev-parse', 'FETCH_HEAD', quiet=True)
+
+    def url(self):
+        return f'https://github.com/WebAssembly/{self.repo}'
+
+    # Helper to run a `git` command within the `repos/spec` repository.
+    def git(self, *args, **kwargs):
+        return git('-C', 'repos/spec', *args, **kwargs)
+
+    # Check out this repository to `repos/spec` with a merge against `spec`
+    # if specified.
+    #
+    # Returns `False` if the merge fails.
+    def checkout_merge(self, spec=None):
+        self.git('checkout', '-B', 'try-merge')
+        self.git('reset', '--hard', self.rev)
+
+        if spec is None:
+            return True
+
+        try:
+            # Attempt to merge with the `spec` repository
+            self.git('merge', '-q', spec.rev, '-m', 'merge')
+        except GitError:
+            # If the merge failed try to ignore merge conflicts in non-test
+            # directories as we don't care about those changes
+            non_tests = ":(exclude)test/"
+            self.git('checkout', '--ours', non_tests)
+            self.git('add', non_tests)
+            try:
+                self.git('-c', 'core.editor=true', 'merge', '--continue')
+            except GitError:
+                # If all that failed then the merge couldn't be done.
+                self.git('merge', '--abort')
+                return False
+        self.merged_rev = self.git('rev-parse', 'HEAD', quiet=True)
+        return True
+
+    # Return a list-of-triples where each triple is
+    #
+    #   (path_to_test, git_path_of_test, destination_path_of_test)
+    #
+    # This is used to run diffs and copy the test to its final location.
+    def list_tests(self):
+        tests = []
+        for subdir in ['core', 'legacy']:
+            for root, dirs, files in os.walk(f'repos/spec/test/{subdir}'):
+                for file in files:
+                    path = os.path.join(root, file)
+                    repo_path = os.path.relpath(path, 'repos/spec')
+                    _, ext = os.path.splitext(path)
+                    if ext != '.wast':
+                        continue
+                    _, dst = os.path.split(path)
+                    if subdir == 'legacy':
+                        dst = 'legacy/' + dst
+                    tests.append((path, repo_path, dst))
+        return tests
+
+
+# Helper to run a `git` command and handle the output/exit code in an ergonomic
+# fashion.
+def git(*args, quiet=False):
+    if not quiet:
+        print('running: git', *args)
+    ret = subprocess.run(["git", *args], stdout=subprocess.PIPE)
```

Comment:
There seems to be a mix of single and double quotes here.  I don't feel strongly which why but I prefer consistency.